### PR TITLE
feat: direction-wise SubGraph RAG

### PR DESCRIPTION
**Pull request by wey-gu**
# Description

Optimization and fixes on KG index retriever and NebulaGraphStore

- Now SubGraph RAG could be **directed**, impl. in NebulaGraph as an example
- **Removed duplicated subject** in the KGIndex KGTableRetriver
- Fixed the object mismatch in NebulaGarph SubGraph retrieval(reported by @BleakStone )
- Optimized Log Printing of KGTableRetriver

## Type of Change

Please delete options that are not relevant.

- [x] Enhancement of existing feature(no break changes)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by wey-gu** [comment]
Now it looks like:



Also, I fixed the issue reported by @BleakStone and some other guys who were trying the RAG demo, where there were cases when the edge was reversed and the object was retrieved as a subject.(subject -[]-**subject**)

File: llama_index/graph_stores/simple.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def _get_rel_map(self, subj: str, depth: int = 2) -> List[List[str]]:
         rel_map = []
         if subj in self.graph_dict:
             for rel, obj in self.graph_dict[subj]:
-                rel_map.append([rel, obj])
+                rel_map.append([subj, rel, obj])
```
Comment (logan-markewich) [inline llama_index/graph_stores/simple.py:49]: Does this change how other graph stores should be building their rel map? (i.e. neo4j, kuzu, falkordb)

Comment (wey-gu) [reply llama_index/graph_stores/simple.py:49]: Yes, previously only simple impl. was not duplicated, the cypher based ones are all with subjects in rels, too(I checked neo4j.)

But I'll double check tomorrow, too.

Comment (wey-gu) [reply llama_index/graph_stores/simple.py:49]: - [ ] neo4j, already with subject
- [ ] kuzu, aha, it's w/o subject, added back
- [ ] falkordb/redisgraph, already with subject

Comment (logan-markewich) [reply llama_index/graph_stores/simple.py:49]: Amazing, thanks for checking!

**Comment by logan-markewich** [comment]
Thanks @wey-gu 🚀

**Comment by wey-gu** [comment]
> Thanks @wey-gu 🚀

Thanks a lot @logan-markewich for the lightning-fast collaboration 😭.

**Comment by radicalloop** [comment]
Thank you!

**Pull request merged**